systemservices/marketplace/src/contracts/utils.ts

```diff
@@ -24,13 +25,11 @@ const fromUnit = (x: string|BigNumber) => new BigNumber(x).dividedBy(1e18)
 
 const parseTimestamp = (x: string) => new Date(new BigNumber(x).times(1000).toNumber())
 
-const hashToHex = (x: string) => {
-  if (x.startsWith('0x')) {
-    return x
-  }
-  return '0x' + x
+const hashToHex = (x: string): string => {
+  if (x.startsWith('0x')) return x
+  return '0x' + base58.decode(x).toString('hex')
```

Comment:
`0x` is a prefix for base16. Function shouldn't print or accept hash starting with `0x` as it's invalid base58

Reply:
I will return an error rather than returning the string 👍

Reply:
Done